GCTC-NTGC/gc-digital-talent
# Accordion refactor

Related to #6040 but does not close it.

## 👋 Introduction

This branch refactors the accordion component to make it possible to created headers needed for #6040 

## 🕵️ Details

- moves the experience accordions out of User Profile to the components directory
- Factors out the current accordion header as a StandardHeader component to maintain current appearance
  - renamed prop from `headerAs` to `headingAs` to reduce confusion :smile: 
- Separates the Accordion Header from the Accordion Trigger so that they can be used independently
- New Accordion Storybook story `WithEditLink` with what will eventually become the new ExperienceHeader

## 🧪 Testing

1. Rebuild app
2. Check that experience accordions look good
3. Check that non-experience accordions look good
  - both regular and simple styles
4. Check Chromatic diffs
5. Check the new Accordion/WithEditLink story

## 📸 Screenshot

![image](https://user-images.githubusercontent.com/8978655/234622504-2f5239c6-4910-410b-8f41-b1fb4cb4ae49.png)
